clang_build/project.py

```diff
@@ -145,8 +145,40 @@ def __init__(self, config, environment, multiple_projects):
             target_node = targets_and_subproject_targets[target_name]
             # Directories
             target_build_dir = self.build_directory if not multiple_targets else self.build_directory.joinpath(target_name)
+            target_root_dir  = self.workingdir
+
+            # If target is marked as external, try to fetch the sources
+            ### TODO: external sources should be fetched before any sources are read in, i.e. even before targets are created
```

Comment:
What are the advantages?

Reply:
The full dependency graph can only be generated once all sources have been downloaded (since any target *may* have further dependencies specified in it's `clang-build.toml`). I believe this may break the discovery of dependencies for nested subprojects (though I cannot think of a specific example of how it could go wrong).